### ISSUE TITLE
Nested single table query forced to be qualified

### ISSUE THREAD

**Issue by KulykDmytro** [sqlfluff/sqlfluff#6251]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

`RF02` [references.consistent] false positives in case of nested single table query
```sql
select *
from (
    select
        a,
        b
    from sch.t1
)
```

### Expected Behaviour

nested single table queries should not be qualified unless specified by `single_table_references`

### Observed Behaviour

```verilog
L:   4 | P:   9 | RF02 | Unqualified reference 'a' found in select with more than one referenced table/view. [references.qualification]
L:   5 | P:   9 | RF02 | Unqualified reference 'b' found in select with more than one referenced table/view. [references.qualification]
```

### How to reproduce

-

### Dialect

athena

### Version

3.2.0

### Configuration

```ini
[sqlfluff:rules:references.consistent]
single_table_references = consistent
# same behaviour for `unqualified`
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by KulykDmytro** [comment]
looks like fixed by #6231
